src/Handler/Search.hs

```diff
@@ -29,17 +28,20 @@ import qualified XMLArrows
 resultsPerPage :: Int
 resultsPerPage = 50
 
+maxPages :: Int
+maxPages = 10
+
 getSearchR :: Handler TypedContent
 getSearchR = do
   mquery <- lookupGetParam "q"
-  page <- maybe 1 (max 1) . ((readMaybe . T.unpack) =<<) <$> lookupGetParam "page"
+  page <- maybe 1 (max 1) . ((readMaybe . unpack) =<<) <$> lookupGetParam "page"
 
   case mquery of
     Nothing -> redirect HomeR
     Just query -> do
       (results, hasMore) <- case tryParseType query of
-        Just ty | not (isSimpleType ty) -> searchForType page ty
-        _ -> searchForName page (toLower query)
+        Just ty | not (isSimpleType ty) -> searchForType (min maxPages page) ty
```

Comment:
I think we should extract the part that makes sure `page <= maxPages` into a separate function at the time we take it out of the query string, together with all of the other validations and things we do to it.